packages/editor-ui/src/components/ModalRoot.vue

```diff
@@ -1,34 +1,23 @@
+<script setup lang="ts">
+import { useUIStore } from '@/stores/ui.store';
+
+const props = defineProps<{
+	name: string;
+	keepAlive?: boolean;
+}>();
+
+const uiStore = useUIStore();
+</script>
+
 <template>
-	<div v-if="uiStore.isModalOpen(name) || keepAlive">
+	<div v-if="uiStore.isModalOpen(props.name) || props.keepAlive">
 		<slot
-			:modal-name="name"
-			:active="uiStore.isModalActive(name)"
-			:open="uiStore.isModalOpen(name)"
-			:active-id="uiStore.getModalActiveId(name)"
-			:mode="uiStore.getModalMode(name)"
-			:data="uiStore.getModalData(name)"
+			:modal-name="props.name"
+			:active="uiStore.isModalActive(props.name)"
+			:open="uiStore.isModalOpen(props.name)"
+			:active-id="uiStore.getModalActiveId(props.name)"
+			:mode="uiStore.getModalMode(props.name)"
+			:data="uiStore.getModalData(props.name)"
```

Comment:
Don't need to reference `props` in `template`. You should be able to use `name` and `keepAlive` directly.

Reply:
Funny enough, if I do that I get the following from VS Code
![CleanShot 2024-06-10 at 16 26 42](https://github.com/n8n-io/n8n/assets/5410822/5c9423b8-99c7-4c4b-812a-9b0eaf5b5eee)

I know it's not a type issue but annoying

Reply:
I had to roll back this change because scoped slots actually hide type issues (they lose type safety, and everything becomes `any`). That's why we didn't have type errors in modal components

With composition API setup script type inference will be correct and we will have a ton of issues I can't fix in this PR
They need to fixed in a separate PR